src/Model/Introduction.php

```diff
@@ -79,43 +76,40 @@ public function confirm(bool $duplex = false, bool $hidden = null)
 			'protocol'  => $protocol,
 			'writable'  => $writable,
 			'hidden'    => $hidden ?? $contact['hidden'],
-			'rel'       => $new_relation,
+			'rel'       => $newRelation,
 		];
 		$this->dba->update('contact', $fields, ['id' => $contact['id']]);
 
 		array_merge($contact, $fields);
 
-		if ($new_relation == Contact::FRIEND) {
+		if ($newRelation == Model\Contact::FRIEND) {
 			if ($protocol == Protocol::DIASPORA) {
-				$ret = Diaspora::sendShare(User::getById($contact['uid']), $contact);
+				$ret = Diaspora::sendShare(Model\Contact::getById($contact['uid']), $contact);
 				$this->logger->info('share returns', ['return' => $ret]);
 			} elseif ($protocol == Protocol::ACTIVITYPUB) {
 				ActivityPub\Transmitter::sendActivity('Follow', $contact['url'], $contact['uid']);
 			}
 		}
 
-		$this->delete();
+		return true;
 	}
 
 	/**
 	 * Silently ignores the introduction, hides it from notifications and prevents the remote contact from submitting
 	 * additional follow requests.
 	 *
-	 * Chainable
-	 *
-	 * @return Introduction
+	 * @return bool
 	 * @throws \Exception
 	 */
 	public function ignore()
 	{
-		$this->dba->update('intro', ['ignore' => true], ['id' => $this->id]);
-
-		return $this;
+		return $this->dba->update('intro', ['ignore' => true], ['id' => $this->id]);
```

Comment:
just a notice: If we inject the Repository, this would maybe possible
```suggestion
		$this->ignore = true;
		return $this->repo->save($this);
```

Reply:
Yes, I see how.